src/nodes.js

```diff
@@ -167,9 +167,7 @@ var getChild = function(parent, key) {
  * @param {!Node} child The child to register.
  */
 var registerChild = function(parent, key, child) {
-  if (key) {
-    getKeyMap(parent)[key] = child;
-  }
+  getKeyMap(parent)[key] = child;
```

Comment:
The type should change from {?string} (a nullable string) to just {string}. Also, the sentence about if a key is provided can be removed.